processor/src/main/java/org/mapstruct/ap/internal/model/ValueMappingMethod.java

```diff
@@ -361,6 +372,7 @@ private boolean reportErrorIfMappedTargetEnumConstantsDontExist(Method method, T
 
             for ( ValueMappingOptions mappedConstant : valueMappings.regularValueMappings ) {
                 if ( !NULL.equals( mappedConstant.getTarget() )
+                    && !THROW_EXCEPTION.equals( mappedConstant.getTarget() )
```

Comment:
I think that we need the same check on line 389 as well. It should be possible to map `ANY_UNMAPPED` or `ANY_REMAINING` into `<THROW_EXCEPTION>`. I know that the second one doesn't make sense as that is the default from MapStruct anyways :)